### PR TITLE
Add filtfilt to cupyx.scipy.signal

### PR DESCRIPTION
**Pull request by andfoy**
See https://github.com/cupy/cupy/issues/7403

This is basically a line-per-line port from SciPy, the only changes are related to the initial state differences between CuPy and SciPy's `lfilter` implementation

### CODE REVIEW

**Comment by takagi** [comment]
Seems that CI makes illegal memory access and fails on CUDA 10.2. I guess some dtype combinations make this problem. Would you try to simply let dtype parametrized instead of `for_all_dtypes_combination`?

File: tests/cupyx_tests/scipy_tests/signal_tests/test_signaltools.py [hunk 1]
```diff
@@ -572,3 +572,81 @@ def test_deconvolve(self, order, xp, scp):
         b = testing.shaped_random((order,), xp, scale=0.3)
         o = scp.signal.convolve(x, b)
         return scp.signal.deconvolve(o, b)
+
+
+@testing.gpu
```
Comment (takagi) [inline tests/cupyx_tests/scipy_tests/signal_tests/test_signaltools.py:577]: This decorator was here due to a historical reason and was removed in #7453.
```suggestion
```

**Comment by andfoy** [comment]
@takagi, the errors are related to some Cython compilation error, do you know if this is a failure on the CIs? The tests are passing just fine in other CUDA versions, as well as my local installation

**Comment by takagi** [comment]
Yes, let us merge this

**Comment by takagi** [comment]
Thanks, @andfoy!

**Pull request merged**